app/src/androidTest/java/com/github/polypoly/app/base/game/PlayerTest.kt

```diff
@@ -0,0 +1,237 @@
+    private val testUser1 = User(42042042, "test_user1", "", emptySkin, zeroStats, listOf(), mutableListOf())
+    private val testUser2 = User(42042043, "test_user2", "", emptySkin, zeroStats, listOf(), mutableListOf())
+    private val testUser3 = User(42042044, "test_user3", "", emptySkin, zeroStats, listOf(), mutableListOf())
+    private val testUser4 = User(42042045, "test_user4", "", emptySkin, zeroStats, listOf(), mutableListOf())
+    private val testUser5 = User(42042046, "test_user5", "", emptySkin, zeroStats, listOf(), mutableListOf())
+    private val testUser6 = User(42042047, "test_user6", "", emptySkin, zeroStats, listOf(), mutableListOf())
+    private val testPlayer1 = Player(testUser1, 100, listOf())
+    private val testPlayer2 = Player(testUser2, 200, listOf())
+    private val testPlayer3 = Player(testUser3, 100, listOf())
+    private val testPlayer4 = Player(testUser4, 0, listOf(), 4)
+    private val testPlayer5 = Player(testUser5, 0, listOf(), 5)
+    private val testPlayer6 = Player(testUser6, 0, listOf(), 5)
+    private val testDuration = 2
+    private val gameRules = GameRules(GameMode.RICHEST_PLAYER, 3, 7,
+        testDuration, 10, LocationRepository.getZones(), 200)
+    private val gameLobby = GameLobby(testUser1, gameRules, "test_game", "123456", false)
+
+    @Before
+    fun startIntents() {
+        gameLobby.addUser(testUser2)
+        gameLobby.addUser(testUser3)
+        gameLobby.addUser(testUser4)
+        gameLobby.addUser(testUser5)
+        gameLobby.addUser(testUser6)
+        gameLobby.start()
+    }
+
+
+    @Test
+    fun compareToConsiderGreaterThePlayerWithMoreMoney() {
+       assertTrue(testPlayer1 < testPlayer2)
+       assertTrue(testPlayer2 > testPlayer1)
+    }
+
+    @Test
+    fun compareToConsiderEqualThePlayerWithSameMoney() {
+        assertTrue(testPlayer1.compareTo(testPlayer3) == 0)
+    }
+
+    @Test
+    fun compareToConsiderSmallerThePlayerWhoLostFirst() {
+        assertTrue(testPlayer4 < testPlayer5)
+    }
+
+    @Test
+    fun compareToConsiderEqualThePlayerWhoLostAtTheSameTime() {
+        assertTrue(testPlayer6.compareTo(testPlayer5) == 0)
+    }
+
+    @Test
+    fun loseMoneyDecreaseTheMoneyOfThePlayer() {
+        val testPlayer = Player(testUser1, 100, listOf())
+        testPlayer.loseMoney(50)
+        assertEquals(50, testPlayer.getBalance())
+    }
+
+    @Test
+    fun loseMoneyDoesNotDecreaseTheMoneyOfThePlayerBelowZero() {
+        val testPlayer = Player(testUser1, 100, listOf())
+        testPlayer.loseMoney(150)
+        assertEquals(0, testPlayer.getBalance())
+    }
+
+    @Test
+    fun loseMoneyThrowsAnExceptionIfTheAmountIsNegative() {
+        val testPlayer = Player(testUser1, 100, listOf())
+        try {
+            testPlayer.loseMoney(-50)
+            fail("Should have thrown an exception")
+        } catch (e: IllegalArgumentException) {
+            assertEquals("The amount of money lost cannot be negative or zero", e.message)
+        }
```

Comment:
You should use assertThrows from junit instead of try/catch

Reply:
fix done